oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LucenePropertyIndex.java

```diff
@@ -1677,6 +1708,28 @@ private List<Facet> getFacetsUncached(int numberOfFacets, String columnName) thr
                 indexNode.release();
             }
         }
+
+        private List<Facet> getFacetsUncached(Facets facets, int numberOfFacets, String columnName) throws IOException {
```

Comment:
`getFacetsUncached(int numberOfFacets, String columnName)` has the exact same logic at lines 1690-1705. What about reducing the code in this class (that is already huge) and call the new `getFacetsUncached` in there?

Reply:
I did this on purpose actually: to keep the old code as is as much as possible, in case we need to switch back to it. I agree it would be better, but hopefully we don't need the old code in the future - in which case we should remove it. I'm fine to create a jira issue about removing the old code in 6 month, so we don't forget. OK?